### ISSUE TITLE
add an option to disable join messages in chat

### ISSUE THREAD

**Issue by henning-schild** [owncast/owncast#1582]
Just hosted an even with v0.0.10 with hundreds of attendees and quite a bit of fluctuation. That all went great! While the name changing notifications are really needed to keep track of who asked a question before changing their name ... the join/leave causes a lot of noise which disturbs the chat flow and makes the chat hard to follow.

It would be nice if such notifications could be disabled in the admin gui.

**Comment by gabek** [comment]
I'm glad it went well! I've been telling people just to hide these messages via CSS, and that's still the solution in the short term. But enough people have asked for the ability to turn off join messages that it seems like it's something that should probably be added.

Renamed the issue to reflect that there are no leave messages.

**Comment by gabek** [comment]
If anybody wants to look at this https://github.com/owncast/owncast/blob/develop/core/chat/server.go#L94 is where to start.

<blockquote><img src="https://repository-images.githubusercontent.com/266445736/4f82fb80-a5b1-11ea-8f1c-69d316f861b5" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast">owncast/server.go at develop · owncast/owncast</a></strong></div><div>Take control over your live stream video by running it yourself.  Streaming + chat out of the box. - owncast/server.go at develop · owncast/owncast</div></blockquote>

**Comment by nebunez** [comment]
I'm happy to take this one, as this is something I want myself. I believe hiding these messages with CSS causes extra messages to be hidden --see #1627

**Comment by cincodenada** [comment]
One thing that I've seen on some IRC clients that I think is useful is that a bunch of joins in a row (without messages in between) are collapsed/summarized as "user1, user2, and 5 other users joined" or some such, with the most recent joins being the ones called out in the text. I think that's a good in between option to consider!

Edit: To give a concrete example (and implementation, albeit in a totally different context), the [irssi-reolving-door](https://github.com/rfreebern/irssi-revolving-door) plugin is where I'm mainly familiar with this.

I'll try to do an example of what this might look like in our context, but don't hold me to it if I screw something up :sweat_smile: 
> somekinda-ocelot joined
> "Hey everybody, this is an actual message"
> other-funky-name joined
> grumpy-clump joined
> other-funky-name is now known as jonny5
> garbanzo-keyboard joined
> sparrow-knuckles joined
> somekinda-ocelot is now known as someguy

I believe Would become something like:
> somekinda-ocelot joined
> "Hey everybody, this is an actual message"
> sparrow-knuckles and three others joined [maybe a mouseover shows "jonny5, garbanzo-keyboard, grumpy-clump"]
> somekinda-ocelot is now known as someguy

**Comment by gabek** [comment]
@nebunez Are you still planning on looking at this? I'm starting on the v0.0.12 items and I want to know if it's something I should take care of.

**Comment by henning-schild** [comment]
@gabek Thanks so much!
